pallene/driver.lua

```diff
@@ -51,32 +52,32 @@ function driver.compile_internal(filename, stop_after)
     local prog_ast
     prog_ast, errs = parser.parse(filename, input)
     if stop_after == "ast" or not prog_ast then
-        return prog_ast, errs
+        return prog_ast, errs, input
```

Comment:
Having everything return an `input` as a third argument smells a bit fishy.

What do you think about splitting `compile_internal` into two functions, one that reads the source code (going up until the part that says "util.get_file_contents") and another one that calls the various compiler steps, starting with parser.parse?